scripts/identify-web-features.js

```diff
@@ -0,0 +1,303 @@
+import { Octokit } from "octokit";
+import { features } from "web-features";
+import yargs from "yargs";
+
+// This is used as a hidden HTML comment when posting comments to GitHub issues.
+// This way, we can retrieve the comment later, and update it if needed.
+const HIDDEN_COMMENT_IN_ISSUE = "<!-- interop-proposals-bot web-features update -->";
+const GITHUB_API_VERSION = "2022-11-28";
+
+const argv = yargs(process.argv)
+  .option("number", {
+    alias: "n",
+    type: "number",
+    default: false,
+    describe: "The issue number to process",
+  })
+  .option("repo", {
+    alias: "r",
+    type: "string",
+    describe: "The owner and repository name. For example: web-platform-tests/interop",
+  }).argv;
+
+const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
+
+function escapeFeatureName(feature) {
+  // Escape the feature name for use in HTML.
+  return feature.name.replace(/</g, "&lt;").replace(/>/g, "&gt;");
+}
+
+async function getReferencedIssue() {
+  const response = await octokit.request(`GET /repos/${argv.repo}/issues/${argv.number}`,);
+  return response.data;
+}
+
+// Identify web-features based on spec URLs in the issue body.
+function gatherFeaturesFromSpecUrls(issueBody, gatheredFeatures) {
+  // Find URLs in the issue body.
+  const urls = issueBody.match(/https?:\/\/[^)\s]+/g) || [];
+
+  // Filter out known non-spec URLs.
+  const specUrls = urls
+    .filter(url => {
+      // Filter out known non-spec URLs.
+      return !url.includes("bugzilla") &&
+        !url.includes("github.com") &&
+        !url.includes("webkit.org") &&
+        !url.includes("developer.mozilla.org") &&
+        !url.includes("developer.chrome.com") &&
+        !url.includes("wpt.fyi") &&
+        !url.includes("css-tricks") &&
+        !url.includes("webstatus.dev") &&
+        !url.includes("learn.microsoft.com") &&
+        !url.includes("chromium.org");
+    }).map(url => {
+      // Separate the # from the URL if it exists.
+      return [url, url.split("#")[0]];
+    });
+
+  // Iterate over web features to find spec URL matches.
+  for (const [candidateUrl, candidateUrlNoAnchor] of specUrls) {
+    for (const id in features) {
+      const feature = features[id];
+      const featureSpecs = Array.isArray(feature.spec) ? feature.spec : [feature.spec];
+      if (featureSpecs.some(spec => spec === candidateUrl || spec === candidateUrlNoAnchor)) {
+        gatheredFeatures.add(id)
```

Comment:
In general I'd prefer that these functions return a set of possible features, rather than editing a set in-place.